spyderlib/plugins/configdialog.py

```diff
@@ -903,77 +917,419 @@ def _save_lang(self):
+        combo_layout.addWidget(schemes_combo.label)
+        combo_layout.addWidget(schemes_combo.combobox)
+
+        buttons_layout = QVBoxLayout()
+        buttons_layout.addLayout(combo_layout)
+        buttons_layout.addWidget(edit_button)
+        buttons_layout.addWidget(self.reset_button)
+        buttons_layout.addWidget(self.delete_button)
+        buttons_layout.addStretch(1)
+        buttons_layout.addWidget(create_button)
+
+        preview_layout = QVBoxLayout()
+        preview_layout.addWidget(self.preview_editor)
+
+        buttons_preview_layout = QHBoxLayout()
+        buttons_preview_layout.addLayout(buttons_layout)
+        buttons_preview_layout.addLayout(preview_layout)
+
+        manage_layout = QVBoxLayout()
+        manage_layout.addWidget(about_label)
+        manage_layout.addLayout(buttons_preview_layout)
+        manage_group = QGroupBox(_("Manage color schemes"))
+        manage_group.setLayout(manage_layout)
+
+        vlayout.addWidget(manage_group)
         self.setLayout(vlayout)
-        
-    @Slot(str)
-    def reset_to_default(self, name):
-        set_default_color_scheme(name, replace=True)
-        self.load_from_conf()
-            
+
+        # Signals and slots
+        create_button.clicked.connect(self.create_new_scheme)
+        edit_button.clicked.connect(self.edit_scheme)
+        self.reset_button.clicked.connect(self.reset_to_default)
+        self.delete_button.clicked.connect(self.delete_scheme)
+        self.schemes_combo.currentIndexChanged.connect(self.update_preview)
+        self.schemes_combo.currentIndexChanged.connect(self.update_buttons)
+
+        # Setup
+        for name in names + custom_names:
+            self.scheme_editor_dialog.add_color_scheme_stack(name)
+
+        self.update_combobox()
+
     def apply_settings(self, options):
         self.main.editor.apply_plugin_settings(['color_scheme_name'])
         if self.main.historylog is not None:
             self.main.historylog.apply_plugin_settings(['color_scheme_name'])
         if self.main.inspector is not None:
             self.main.inspector.apply_plugin_settings(['color_scheme_name'])
+        self.update_preview()
+        self.update_combobox()
+
+    # Helpers
+    # -------------------------------------------------------------------------
+    @property
+    def current_scheme(self):
+        return self.schemes_combo.currentText().lower()
+
+    @property
+    def current_scheme_index(self):
+        return self.schemes_combo.currentIndex()
+
+    def update_combobox(self):
+        """Recreates the combobox contents."""
+        index = self.current_scheme_index
+        self.schemes_combo.blockSignals(True)
+        names = self.get_option("names")
+        custom_names = self.get_option("custom_names")  # 'custom-n'
+
+        if custom_names:
+            choices = names + [None] + sorted(custom_names)
+            scheme_choices = list(zip(choices, [a for a in choices]))

```

Comment:
Why using the cumbersome `[a for a in choices]` instead of just `choices`? It's the same thing :-) (unless you've forgotten something in the list comprehension).

I think this code (as well as the one below) can be simplified by using a dict comprehension, like this

```python
{c: c.lower() for c in choices}
```

because it seems you want to create a mapping between the scheme names in upper and lower case.